### PR TITLE
hub: add support to send user commands

### PR DESCRIPTION
**Pull request by wlozano0collabora**
Please check carefully this approach as I'm not sure this is the right one.
I added the support to send user commands as AMQP messages in hub, as previously the support was only for incoming ones.

### CODE REVIEW

File: chamge/amqp-hub-backend.c [hunk 1]
```diff
@@ -887,6 +924,141 @@ chamge_amqp_hub_backend_deactivate (ChamgeHubBackend * hub_backend)
...
+
+  if (amqp_parse_url (amqp_uri, &connection_info) != AMQP_STATUS_OK) {
+    g_set_error_literal (error, CHAMGE_BACKEND_ERROR,
+        CHAMGE_BACKEND_ERROR_INVALID_PARAMETER, "url pasring failure");
+    goto out;
+  }
+
+  amqp_channel = g_settings_get_int (self->settings, "amqp-channel");
+  amqp_exchange_name =
+      g_settings_get_string (self->settings, "enroll-exchange-name");
```
Comment (xhaakon) [inline chamge/amqp-hub-backend.c]: @justinjoy @hkseo95 Is it correct to use "enroll-exchange-name" here? Isn't it something strictly related to "enroll" command?

File: chamge/amqp-hub-backend.c [hunk 2]
```diff
@@ -104,6 +104,43 @@ _amqp_rpc_reply_string (amqp_rpc_reply_t r)
   }
 }
 
+inline static const gchar *
+_amqp_get_rpc_reply_string (amqp_rpc_reply_t r)
```
Comment (xhaakon) [inline chamge/amqp-hub-backend.c]: I see we're duplicating a lot of code between the different AMQP backends. There's definitely a lot to be improved in the matter of maintainability of Chamge code. I know we're about to replace rabbitmq with another library; it's just something to have in mind when we're re-implementing this.

File: chamge/amqp-hub-backend.c [hunk 3]
```diff
@@ -887,6 +924,141 @@ chamge_amqp_hub_backend_deactivate (ChamgeHubBackend * hub_backend)
...
+      goto out;
+    }
+  }
+
+out:
+  return ret;
+}
+
+
+
```
Comment (xhaakon) [inline chamge/amqp-hub-backend.c]: nitpick: Separate functions with one empty line. You tend to use 1-3 at random.

File: chamge/amqp-hub-backend.c (hunk 2 above)
Comment (wlozano0collabora) [reply chamge/amqp-hub-backend.c]: Yes, it is the same issue I pointed but @justinjoy said

> It has some reason to have duplicate code in amqp-*-backend;
> 
>  * we didn't sure librabbitmq-c is stable or production-ready
>  *  we tended to implement each backend by a different developer, and different style to test librabbitmq-c in various ways while implementation
>  *  but, finally, lack of time and developer. As a result, we do just copy-n-paste
> 
> Having a common set is good idea, but now modifying chamge-arbiter and chamge-edge to have a common function set is a very low priority.
> One option to raise priority the task is only if you find a severe bug while handling chamge-hub APIs.

File: chamge/amqp-hub-backend.c (hunk 3 above)
Comment (wlozano0collabora) [reply chamge/amqp-hub-backend.c]: Sorry, you are right. As probably @justinjoy will require some changes I can fix it in the next version.

File: chamge/amqp-hub-backend.c (hunk 1 above)
Comment (hkseo95) [reply chamge/amqp-hub-backend.c]: > @justinjoy @hkseo95 Is it correct to use "enroll-exchange-name" here? Isn't it something strictly related to "enroll" command?

Yes, it is corrent. We wanted  this to change in gsettings. "enroll-exchange-name" is exchange name for enroll.

**Pull request merged**